src/group_theory/perm/sign.lean

```diff
@@ -169,20 +173,20 @@ lemma swap_mul_eq_iff {i j : α} {σ : perm α} : swap i j * σ = σ ↔ i = j :
 lemma is_swap_of_subtype {p : α → Prop} [decidable_pred p]
   {f : perm (subtype p)} (h : is_swap f) : is_swap (of_subtype f) :=
 let ⟨⟨x, hx⟩, ⟨y, hy⟩, hxy⟩ := h in
-⟨x, y, by simp at hxy; tauto,
+⟨x, y, by simp only [ne.def] at hxy; tauto,
   equiv.ext $ λ z, begin
     rw [hxy.2, of_subtype],
-    simp [swap_apply_def],
+    simp only [swap_apply_def, coe_fn_mk, swap_inv, subtype.mk_eq_mk, monoid_hom.coe_mk],
     split_ifs;
-    cc <|> simp * at *
+    rw subtype.coe_mk <|> cc,
```

Comment:
There's 6 goals here, goals 1,2 and 3 are solved by the `rw`, goals 4,5,6 are solved by the `cc`. Is there a better way to write this? (maybe just write out all 6 cases?)

Reply:
actually one of the cases solved by `cc` is just `refl`.

Reply:
I think this is fine. The first two are reasonable places to use `cc`, even if it's a little slow. I think being concise is worth a few milliseconds here.